From 7011178c336f15f9d68f6791d7e8ac96bcba676c Mon Sep 17 00:00:00 2001
From: Attila Fidan <dev@print0.net>
Date: Sun, 20 Apr 2025 11:30:03 +0000
Subject: [PATCH 025/166] winewayland: Require wl_pointer for pointer
 constraints.

If ClipCursor is called while the seat doesn't have a pointer and the
call qualifies for locking the pointer, it would have tried to lock a
null wl_pointer.
---
 dlls/winewayland.drv/wayland_pointer.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_pointer.c b/dlls/winewayland.drv/wayland_pointer.c
index b2af6c0db4a..257735dc909 100644
--- a/dlls/winewayland.drv/wayland_pointer.c
+++ b/dlls/winewayland.drv/wayland_pointer.c
@@ -710,9 +710,10 @@ static void wayland_pointer_update_constraint(struct wl_surface *wl_surface,
     }
 
     needs_lock = wl_surface && (((confine_rect || covers_vscreen) &&
-                 !pointer->cursor.wl_surface) || force_lock);
+                 !pointer->cursor.wl_surface) || force_lock) &&
+                 pointer->wl_pointer;
     needs_confine = wl_surface && confine_rect && pointer->cursor.wl_surface &&
-                 !force_lock;
+                 !force_lock && pointer->wl_pointer;
 
     if (!needs_confine && pointer->zwp_confined_pointer_v1)
     {
-- 
2.50.0

